### ISSUE TITLE
Not able to find the build steps for Linux

### ISSUE THREAD

**Issue by LalatenduMohanty** [telegramdesktop/tdesktop#880]
Hi I can't find any build steps for Linux e.g. Fedora. If I can get the build steps, I can try package the tdesktop as a RPM in Fedora.

**Comment by Aokromes** [comment]
I think you are looking for:
https://github.com/telegramdesktop/tdesktop/blob/master/QTCREATOR.md

**Comment by hobarrera** [comment]
Those instructions are not useful for packagers, since an IDE/GUI needs human interaction, and cannot be scripted easily.

The [PKGBUILD](https://aur.archlinux.org/packages/te/telegram-desktop-git/PKGBUILD) for ArchLinux might be a useful starting point. The format is rather simple, so you should find out the steps to programatically build telegram-desktop there. Functions are called in-order `prepare`, `build`, `package`.

**Comment by LalatenduMohanty** [comment]
Thanks @hobarrera . I think it has most of the information I am looking for.

**Comment by hobarrera** [comment]
You're welcome. It'd be great to have this in the readme or the docs somewhere, if you've the time. :)

**Comment by bil-elmoussaoui** [comment]
@auchri you can close this too ;)

**Comment by hobarrera** [comment]
@bil-elmoussaoui 

While the link I provided answered the original question, I think that this information should be incorporated into the readme or the documentation before closing the ticket. The ticket is not just about answering to this one person, but filling in missing bits of documentation.

**Comment by bil-elmoussaoui** [comment]
Building for fedora is not different from building in Ubuntu ;) and this information is provided in Readme, i don't think that telegram should add every distro name in the readme file

**Comment by hobarrera** [comment]
@bil-elmoussaoui 
Have you followed up on the comments on this issue? The question is not Fedora-specific, or Ubuntu-specific, and the README does not provide an answer to it for *any* distribution.

Note that the link I provided was for the ArchLinux package, yet served as a reference for Fedora (and works for Ubuntu too).

The critical thing here is that the README provides details for how **developers** should be building tdesktop, but no clear clues for packages (and let's be honest, packagers are the ones who will need this the most).

**Comment by auchri** [comment]
So feel free to write instructions and open a pull request :)

**Comment by bil-elmoussaoui** [comment]
Well, i'm not a collaborator or a tdesktop dev. i was just helping closing duplicate/fixed issue to make this repo cleaner. And the best way is to ask Telegram to build a version for arch, fedora and debian and keep the users far from any copy of Telegram(a modified one)

**Comment by hobarrera** [comment]
> Well, i'm not a collaborator or a tdesktop dev. i was just helping closing duplicate/fixed issue to make this repo cleaner. And the best way is to ask Telegram to build a version for arch, fedora and debian and keep the users far from any copy of Telegram(a modified one)

I don't quite understand what you mean, sorry, can you rephrase?

Again, this is not about distro-specific intructions, on the contrary, it's about distro-agnostic instructions for packagers to compile tdesktop for different distributions themselves.
